### PR TITLE
FW-104

### PR DESCRIPTION
**Pull request by coakenfold**
These pages have been updated to show a message if an anon user tries to access them:
- Dialect/Portal Edit
- Word
- Word - Edit
- Phrase
- Phrase - Edit
- Songs
- Songs - Edit
- Stories
- Stories - Edit
- Gallery
- Gallery - Edit
- Phrasebook
- Phrasebook - Edit
- Contributor
- Contributor - Edit
- Recorder
- Recorder -Edit

The changes in: https://github.com/First-Peoples-Cultural-Council/fv-web-ui/pull/84/commits/d0eb563bb6b1740e347a2c3db96c2e7a1dba18ce are a good example of what has been done across the site & in this PR (though it is missing the `<PromiseWrapper />` component which has been added in other components).

### CODE REVIEW

File: frontend/app/assets/javascripts/views/pages/explore/dialect/learn/songs-stories/edit.js [hunk 1]
```diff
@@ -76,58 +83,90 @@ export class PageDialectBookEdit extends Component {
...
+    editPageDialogOpen: false,
+    editPageItem: null,
+    formValue: null,
+    sortedItems: List(),
+    componentState: STATE_LOADING,
   }
 
   // Redirect on success
-  componentWillReceiveProps(nextProps) {
+  componentDidUpdate(prevProps) {
```
Comment (coakenfold) [inline frontend/app/assets/javascripts/views/pages/explore/dialect/learn/songs-stories/edit.js]: FW-233

**Comment by dy2288** [comment]
Question @coakenfold :

Is the intention to have internationalization.js within components override the ErrorBoundary internationalization.js file? I tested by removing the `copy -> errorBoundary -> title` property in `words->Edit->internationalization.js`, however `We encountered a problem` did not show up, it was just blank.

Also see comment on FW-104 for wording and 1 next step.

**Comment by coakenfold** [comment]
> Is the intention to have internationalization.js within components override the ErrorBoundary internationalization.js file? I tested by removing the `copy -> errorBoundary -> title` property in `words->Edit->internationalization.js`, however `We encountered a problem` did not show up, it was just blank.

Yes it is. The idea being we could customize the title to whatever is being created, eg: 'We encountered a problem creating the recorder'. 

However, the way the code is setup currently it is an all or nothing prospect. If `copy` prop is not set on `<ErrorBoundary />` it will then dynamically import `ErrorBoundary/internationalization.js` and use it as a fallback.

I'll see about making the determination a bit smarter. Perhaps looking for this.props.copy.errorBoundary to decide to dynamically import?

The code in question:
```
  async componentDidMount() {
    const copy =
      this.props.copy ||
      (await import(/* webpackChunkName: "ErrorBoundaryInternationalization" */ './internationalization').then(
        (_module) => {
          return _module.default
        }
      ))
    this.setState({
      copy,
    })
  }
```

**Comment by dy2288** [comment]
> I'll see about making the determination a bit smarter. Perhaps looking for this.props.copy.errorBoundary to decide to dynamically import?

Not a bad idea, but we can deal with it when the scenario comes up.

**Comment by coakenfold** [comment]
Ah sorry, found some bugs in `FW-104`.

Please hold off on the merge to `master`, looking into the issue now.

**Pull request closed without merging**